Dockerfile

```diff
@@ -9,20 +9,24 @@ WORKDIR /app
 
 COPY poetry.lock /app/
 COPY pyproject.toml /app/
-COPY ${package}/ /app/${package}
 
-RUN python3.9 -m pip install poetry \
- && python3.9 -m poetry config virtualenvs.create false \
- && python3.9 -m poetry install --without dev \
- && python3.9 -m poetry export -f requirements.txt --output requirements.txt --without-hashes
+RUN python3.9 -m pip install poetry==1.4.2
+RUN python3.9 -m poetry config virtualenvs.create false
+RUN python3.9 -m poetry install --without dev --no-root
+RUN python3.9 -m poetry export -f requirements.txt --output requirements.txt --without-hashes
```

Comment:
Here each RUN adds a new layer to the image. I don't see a reason for using multiple RUINs

Reply:
Agreed with Sandro, here's how a chained `RUN` might look:

```Dockerfile
RUN pip3 install "poetry==1.4.2" \
 && poetry config virtualenvs.create false \
 && poetry install --without dev --no-root \
 && poetry export -f requirements.txt --output requirements.txt --without-hashes
```

Reply:
Btw, I omitted the `python3.9` in all of these I'm curious why they were there in the first place. Does the base image have more than one Python3 install?

Reply:
Ack; I removed the chained command and replaced it with the multiple runs as more readable, but I didn't realize it would cause more layers. I'll fix that.

The python version in the command looks extraneous, but it follows the general rule of being (overly) explicit to avoid drift. It probably doesn't matter here; it's just extra cautious.